### PR TITLE
Blog post: Use Camel to integrate multiple systems with a LLM

### PR DESCRIPTION
**Pull request by Croway**
Based on https://github.com/Croway/camel-whatsapp-chatbot

### CODE REVIEW

File: content/blog/2024/02/camel-whatsapp-langchain4j/index.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+The objective is the following, I'd like to have specific conversations about some topic, in this case, how to contribute to Apache Camel, with an LLM via WhatsApp. In this context WhatsApp is just an example, Apache Camel offers 300+ components that can be easily integrated!
+
+
+The objective is clear, but what about the implementation? Libraries like langchain4j and Apache Camel help a lot with this kind of use case, in particular, we will leverage the following features:
+* camel-whatsapp will take care of the integration with WhatsApp APIs and thanks to the camel-webhook feature the communication with the WhatsApp APIs is effortless.
+* On the other hand, langchain4j offers abstractions and toolkits that help developers interact with LLMs.
+
+
+In this example, the model [GPT-3.5 Turbo](https://platform.openai.com/docs/models/gpt-3-5-turbo) is used, and the [camel core contribution documentation](https://camel.apache.org/camel-core/contributing/) is used as an embedding, in this way it will be possible to have clear conversations about camel contributions.
```
Comment (oscerd) [inline content/blog/2024/02/camel-whatsapp-langchain4j/index.md:25]: For the camel link you need to use the xref syntax

**Review by oscerd — CHANGES_REQUESTED** [review]
Just a minor one

**Review by oscerd — APPROVED** [review]

**Review by davsclaus — APPROVED** [review]

**Review by orpiske — APPROVED** [review]

File: content/blog/2024/02/camel-whatsapp-langchain4j/index.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
+---
+title: "Integrate your AI models effortlessly with Apache Camel"
+date: 2024-02-09
```
Comment (apupier) [inline content/blog/2024/02/camel-whatsapp-langchain4j/index.md:3]: I guess you meant 2024-02-19

File: content/blog/2024/02/camel-whatsapp-langchain4j/index.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+draft: true
+authors: [croway]
+categories: ["Camel Spring Boot", "Usecases", "AI"]
+preview: "Build a conversational AI integration with Apache Camel, langchain4j, and WhatsApp."
+---
+
+
+This blog shows how Apache Camel can help integrate multiple systems with an AI model, in particular, the [camel-whatsapp](/components/4.0.x/whatsapp-component.html) component is used to build a chat on WhatsApp; so that a user can easily communicate with the LLM via WhatsApp.
```
Comment (apupier) [inline content/blog/2024/02/camel-whatsapp-langchain4j/index.md:11]: I think you need to define the acronym LLM as first occurence (or provide a link)

File: content/blog/2024/02/camel-whatsapp-langchain4j/index.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+[4] The `bean(agent)` invokes the LLM with the message sent by the user, that we set into the Camel body in [3] once the computation is done, the response message from the LLM is set into the Camel body.
+
+
+[5] `.convertBodyTo(TextMessageRequest.class)` is implemented by a Camel TypeConverter that takes a String message and the Exchange as input and creates a `TextMessageRequest` object that can be serialized and sent to the WhatsApp API.
+
+
+[5] And finally, `.to(whatsapp:{{camel.component.whatsapp.phone-number-id}})` we reply to the phone number that initiated the conversation.
+
+
+# Conclusions
```
Comment (apupier) [inline content/blog/2024/02/camel-whatsapp-langchain4j/index.md:145]:
```suggestion
# Conclusion
```

**Comment by Croway** [comment]
@apupier thanks a lot for the comments, I implemented them, and added a small section about the TypeConverters (fyi @davsclaus)

File: content/blog/2024/02/camel-whatsapp-langchain4j/index.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
+---
+title: "Integrate your AI models effortlessly with Apache Camel"
+date: 2024-02-19
+draft: true
```
Comment (apupier) [inline content/blog/2024/02/camel-whatsapp-langchain4j/index.md:4]: I'm not sure on the impact of the `draft` attribute.
When ready to publish I guess it needs to be turned to false

**Review by apupier — COMMENTED** [review]
I think that adding a video or an animated image demoing the final result would improve the wow effect and ease the understanding of the use case. BUT I let you judge if it is worthy to spend the time to do it.

**Comment by Croway** [comment]
> I think that adding a video or an animated image demoing the final result would improve the wow effect and ease the understanding of the use case. BUT I let you judge if it is worthy to spend the time to do it.

Do you know if I can load a video on camel-website? anyway, the video con be found on the example repo https://github.com/croway/camel-whatsapp-chatbot

**Comment by Croway** [comment]
@apupier I added a demo as gif, thanks for the feedback!

**Comment by apupier** [comment]
> > I think that adding a video or an animated image demoing the final result would improve the wow effect and ease the understanding of the use case. BUT I let you judge if it is worthy to spend the time to do it.
> 
> Do you know if I can load a video on camel-website? anyway, the video con be found on the example repo https://github.com/croway/camel-whatsapp-chatbot

i do not know to embed a video but here it is pointing to a youtube video https://github.com/apache/camel-website/blob/6085d8f3c1904c2d56b7170abf231bdf57e73a42/content/blog/2023/12/camel-quarkus-native-vscode-debug/index.md?plain=1#L18

**Pull request merged**